.gitlab-ci-check-docker-build.yml

```
@@ -187,3 +191,38 @@ trigger:saas:sync-staging-component:
           -F variables[SYNC_IMAGE_TAG]=${DOCKER_PUBLISH_COMMIT_TAG}
           https://gitlab.com/api/v4/projects/17809112/trigger/pipeline
     - done
+
+# saas-specific job to retag docker images after saas-* tags are pushed to the repository
+publish:image:saas:
+  stage: publish
+  only:
+    refs:
+      - /^saas-[a-zA-Z0-9.]+$/
+  tags:
+    - docker
+  image: docker
+  services:
+    - docker:19.03.5-dind
+  dependencies:
+    - build:docker
+  before_script:
+    - export SOURCE_TAG=staging_${CI_COMMIT_SHA}
+    - export DOCKER_TAG=${CI_COMMIT_REF_SLUG}
+    - export SERVICE_IMAGE=${DOCKER_REPOSITORY}:${DOCKER_TAG}
```

To be consistent with the rest of the pipeline, I suggest renaming like:

```suggestion
    - export SOURCE_TAG=staging_${CI_COMMIT_SHA}
    - export DOCKER_PUBLISH_TAG=${CI_COMMIT_REF_NAME}
    - export SERVICE_IMAGE=${DOCKER_REPOSITORY}:${DOCKER_PUBLISH_TAG}
```

Using `CI_COMMIT_REF_SLUG` would work as we are planning the tags, but if we ever have incompatible tags, I rather let it fail and not use that tags.

---

Done